ckanext/ontario_theme/plugin.py

```diff
@@ -394,6 +394,202 @@ def get_keyword_count(keyword_lang, language):
+    only the paginated number of packages for the current page 
+    is available, which prohibits application of any custom 
+    sorting since sorting must be done on the full list before
+    pagination.
+
+    q
+        Search query. Passed through c.q.
+    
+    request_params_items
+        Contains the facet search parameters. Passed through
+        request.params.
+
+    current_org
+        When datasets are accessed by clicking on an Organization,
+        the package search must be limited to the current
+        organization, which is stored in this variable. This 
+        variable is not needed when accessing datasets through
+        the Datasets page.
+
+    current_group
+        When datasets are accessed by clicking on a Group,
+        the package search must be limited to the current
+        group, which is stored in this variable. This 
+        variable is not needed when accessing datasets through
+        the Datasets page.
+    
+    item_count
+        Number of packages matching search. Passed through 
+        c.page.item_count.
+
+    '''
+    q = kwargs['q']
+    request_params_items = kwargs['request_params_items']
+    current_org = kwargs['current_org']
+    current_group = kwargs['current_group']
+    item_count = kwargs['item_count']
+
+    # Excerpt from search() fn in ckan/ckan/controllers/package.py
+    # Needed to collect search parameters from facets.
+    search_extras = {}
+    fq = ''
+    for (param, value) in OrderedDict(request_params_items).items():
+        if param not in ['q', 'page', 'sort'] \
+            and len(value) and not param.startswith('_'):
+            if not param.startswith('ext_'):
+                fq += ' %s:"%s"' % (param, value)
+            else:
+                search_extras[param] = value
+    
+    # Add organization to facet query
+    # see e.g. https://localhost/api/action/package_search?fq=organization:helloworld&include_private=true
+    if current_org:
+        fq += ' %s:"%s"' % ('organization', current_org)
+    
+    # Add groups to facet query
+    # see e.g. https://localhost/api/action/package_search?fq=groups:2019-novel-coronavirus
+    if current_group:
+        fq += ' %s:"%s"' % ('groups', current_group)
+
+    # Get the full set of packages returned by search query q
+    # and any facet queries fq.
+    # Note that number of packages matched will be limited to 10
+    # unless otherwise specified in 'rows'. Since we already know
+    # there are 'item_count' number of matches, we can set rows 
+    # to 'item_count'.
+    package_search=toolkit.get_action('package_search')(
+                data_dict={
+                        'q': q,
+                        'fq': fq.strip(),
+                        'start': 0,
+                        'sort': 'title desc',
+                        'rows': item_count,
+                        'extras': search_extras,
+                        'include_private': True
```

Comment:
> private datasets don't show up in any search

Doesn't this line in the code negate this ^

Reply:
@Alexandra-Green I tested and private files are not shown to users without access. Can you also check?

I believe that this line tells the search to include them (otherwise they will be ignored), but for display on the page it is checked elsewhere if the user has access.